pkg/sql/conn_executor_exec.go

```diff
@@ -859,9 +859,20 @@ func (ex *connExecutor) makeExecPlan(ctx context.Context, planner *planner) erro
 		return err
 	}
 
+	flags := planner.curPlan.flags
+
+	// We don't execute the statement if:
+	// - plan contains a full table or full index scan.
+	// - the session setting disallows full table/index scans.
+	// - the query is not an internal query.
+	if (flags.IsSet(planFlagContainsFullIndexScan) || flags.IsSet(planFlagContainsFullTableScan)) &&
+		planner.EvalContext().SessionData.DisallowFullTableScans && ex.executorType == executorTypeExec {
+		return errors.New("full table/index scans are not allowed")
```

Comment:
pgerror

maybe include the query in the error message?

The message should be more descriptive -- needs to include we are rejecting this query because
* it contains full scan
* and the following cluster setting is set